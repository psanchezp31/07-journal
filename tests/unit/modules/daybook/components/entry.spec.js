import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en el entry component", () => {
  // mockrouter
  const mockRouter = {
    push: jest.fn(), //mock a function
  };
  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[1], //simulamos que esta es la nueva entry
    },
    global: {
      mocks: {
        $router: mockRouter, // se hace un obj mock para simular los objetos globales como $router y que apunte a la funcion mockiada
      },
    },
  });

  test("debe de hacer match con el snapshot ", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de redireccionar al hacer click en el entry-container", () => {
    wrapper.find(".entry-container").trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: journalState.entries[1].id }, //verificar que haya sido llamado con este param
    });
  });

  test("pruebas en las propiedades computadas", () => {
    //wrapper.vm. .<--. ver las prop computadas
    //verificar day, month, yearDay
    expect(wrapper.vm.day).toBe(14);
    expect(wrapper.vm.month).toBe("Febrero");
    expect(wrapper.vm.yearDay).toBe("2022, Lunes");
  });
});
