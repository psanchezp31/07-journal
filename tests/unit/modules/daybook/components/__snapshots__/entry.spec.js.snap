// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry component debe de hacer match con el snapshot  1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">14</span><span class="mx-1 fs-5">Febrero</span><span class="mx-1 fw-light">2022, Lunes</span></div>
  <div class="entry-description">Mannyto </div>
</div>
`;
